controllers/auth.js

```diff
@@ -51,14 +51,44 @@ module.exports = {
     };
   },
   register: async (req, res) => { 
-    try { 
-      // TODO: User registration
 
-    } catch (error) { 
-      return res.status(500).json({ 
-        message: error,
-      });
-    };
+    // creates uuid to use as user's ref number with the help of uuid npm
+    const uuid = uuidv4();
+    const { first_name, last_name, email, password } = req.body
+
+    // saltRound for bcrypt
+    const saltRounds = 10;
+
+    //checks if user already exists
+    connection.query('select * from users where email = ?', [email], (err, results) => {
+      // if no user exists create hashed password and add to user table
+      if (results[0] === undefined) {        
```

Comment:
I believe this should just be `results === undefined`. If the results are undefined then there wouldn't be a [0] position to pull from and as a result could cause the app to fail without a try/catch statement.

Reply:
In testing the results that were returned from the initial query returned `[]` so setting this length check allowed it to go through the process and save the user. Otherwise I kept reaching the duplicate email error.

Reply:
I had initially put "results.length === 0" and for some reason it wasn't working correctly for me. I must have had something else incorrect at the time because it definitely works now.  Initially, I was comfortable using the resutls[0] because no matter what you got a results array back from the db but I do prefer the  "results.length === 0" approach. Do we want to have it keep going if no results come back from the db search? would that meant the db is down?  Should I add in a try/catch statement instead?

Reply:
I haven't tested it enough to know, but I feel like I have either found `[]` or `undefined` in the response. Thinking more about this now though, I get the feeling that `undefined` happened when I had not included the correct DB login information in the .env file. 

Perhaps we can test? If that is correct, I think `|| results === undefined` can be removed as we would never have been able to update if the DB connection was down and as a result should stop the call and have it throw the 500 instead of attempting to update a database that we were never connected to.

Outside of this, when `results[0]` was in there I got a `[0] TypeError: Cannot read property '0' of undefined` which prompted me to change it to .length (though I now suspect that this is because of the lack of database connection). If we can confirm that `results[0]` always returns `undefined` when there is no 0 position in an array then I'm all for keeping it as such!